foreshadow/transformers/core/wrapper.py

```diff
@@ -104,27 +90,47 @@ def make_pandas_transformer(transformer):
     ..#noqa: I401
 
     """
-    # use the same base
-    # metaclass as the transformer, otherwise we will get MRO metaclass
-    # issues in DFTransformer if we try to choose the base class for our
-    # metaclass that is not the same one for the transformer we are also
-    # extending.
+    # use the same base metaclass as the transformer, otherwise we will get
+    # MRO metaclass issues in DFTransformer if we try to choose the base class
+    # for our metaclass that is not the same one for the transformer we are
+    # also extending.
     class DFTransformerMeta(type(transformer)):
         """Metaclass for DFTransformer to appear as parent Transformer."""
 
-        def __new__(mcs, *args, **kwargs):
-            class_ = super(DFTransformerMeta, mcs).__new__(
-                mcs, *args, **kwargs
-            )
-            class_.__name__ = transformer.__name__
-            class_.__doc__ = transformer.__doc__
+        def __new__(meta, name, bases, class_dict):
+            class_ = super().__new__(meta, name, bases, class_dict)
+
+            if name == "DFTransformer":
+                # If directly overriding a transformer using magic imports or
+                # using decorator, imitate the parent class and register as the
+                # parent class.
+                # TODO (@Chris): The list of magic methods that are mapped
+                # might need to increase.
+                name_ = transformer.__name__
+                class_.__name__ = name_
+                class_.__doc__ = transformer.__doc__
+            else:
+                # If not directly wrapped, make sure to register the name of
+                # the actual class being wrapped
+                name_ = name
+
+            # Only serialize if directly inheriting from SerializerMixin
+            if SerializerMixin in bases:
+                register_transformer(class_, name_)
```

Comment:
I don't think we should need a registry, either the object extends from SerializerMixin and therefore can be (de)serialized or it doesn't and can't

Reply:
https://github.com/georgianpartners/foreshadow/pull/96/files/ad3ba4fce4ba53df429398bd41bea0596ee06a34?file-filters%5B%5D=.csv&file-filters%5B%5D=.json&file-filters%5B%5D=.lock&file-filters%5B%5D=.pkl&file-filters%5B%5D=.py#r304019512